soundfile.py

```diff
@@ -1015,6 +1038,13 @@ def write(self, data):
 
         """
         import numpy as np
+
+        if self._compression_level is not None:
+            # needs to be called before set_bitrate_mode
+            self._set_compression_level(self._compression_level)
```

Comment:
These calls should happen in the SoundFile constructor only. It is probably not safe to call them anywhere else, as they presumably can't be modified in an already-opened file.

Reply:
I set it just before writing to make sure it works in modes like w+.
All tests are OK when called it at the end of the constructor, any concerns about calling it here?
```diff
    def __init__(...):
        ...
        self._file = self._open(file, mode_int, closefd)
        if set(mode).issuperset('r+') and self.seekable():
            # Move write position to 0 (like in Python file objects)
            self.seek(0)
        _snd.sf_command(self._file, _snd.SFC_SET_CLIPPING, _ffi.NULL,
                        _snd.SF_TRUE)
+       if self._compression_level is not None:
```

Reply:
That looks like the right place